Vagrant! You know, for development

amygori opened the pull request:
Needs testing, but I have a working Vagrant box for local development on my machine...see if it works for you!

I think the provisioner has some unnecessary stuff lingering in there, so that may need updates, but I wanted to get this out for some field testing asap.

Closes #439

amygori committed:
Vagrantfile and shell script provisioner: first pass

-- also add .vagrant directory to .gitignore

amygori committed:
Specify bash shell for Vagrant environment

amygori committed:
Add port forwarding to Vagrantfile

amygori committed:
add postgres version and other dependencies to provisioner script

amygori committed:
Vgrantfile and shell script provisioner updates

amygori committed:
Use port 3000

amygori committed:
Working Vagrantfile!

amygori committed:
Vagrant documentation temporarily in README.md

kstack7 commented:
Woo! I'm so excited to try this :) Thank you so much, @amygori :D

DBNess commented:
🙌😍
On Jun 15, 2016 11:36 PM, "Kristen" notifications@github.com wrote:

> Woo! I'm so excited to try this :) Thank you so much, @amygori
> https://github.com/amygori :D
> 
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> https://github.com/girldevelopit/gdi-website/pull/461#issuecomment-226380255,
> or mute the thread
> https://github.com/notifications/unsubscribe/AANztKHkevo4S5X1akqmt_Trhav6B9YNks5qMMSlgaJpZM4I280m
> .

kstack7 commented on README.md:
```diff
@@ -13,3 +13,19 @@ Girl Develop It is a nonprofit organization that exists to provide affordable an
+6. $ cd /opt/gdi/development
+7. $ bundle install
+8. $ rake db:create db:migrate db:seed
+9. $ bin/rails s
```
I used the "rails s" command to get it running...so just nix bin (this is more of a note for when move this to actual documentation)

kstack7 commented on README.md:
```diff
@@ -13,3 +13,19 @@ Girl Develop It is a nonprofit organization that exists to provide affordable an
+1. [Install VirtualBox](https://www.virtualbox.org/)
+2. [install Vagrant](https://www.vagrantup.com/downloads.html)
+3. [clone the git repository](https://github.com/girldevelopit/gdi-website.git)
+4. $ vagrant up
```
We will just want to add that this is a big setup process so to give at least 5 min to see things pull through...

kstack7 commented:
THIS IS EXCITING!! Total success on my end :)

![vagrant](https://cloud.githubusercontent.com/assets/6612770/16106957/ee1c2116-3366-11e6-9a77-47335450e594.png)

amygori commented:
@kstack7 I am actually floored that it worked perfectly for you!!! SO GREAT! Are you on a PC or Mac? So much agree, the Ruby compilation step should be documented as taking a while. I made that tradeoff to avoid having to use rbenv or rvm to install Ruby. I can update the note in the provisioner to be more than just "a few minutes" but something more visible as a heads up in the documentation would also be good.

kstack7 commented:
@amygori yup! It was very smooth...the load time is totally worth it! I did this on my project-free PC so no chance of conflicts or existing library fun. I can try on my other dev PC and my Mac to help :) So far, so great!

amygori commented:
EXCELLENT. I will let out a big sigh of relief once @Roenok can get it up and running.

imuchnik commented:
OMG OMG OMG ! I am going to try it and let you know. This is awesome!

Roenok commented:
I'll try tonight! I've got some nice, clean virtual machines I can use to test in different environments

kstack7 commented:
Two for two! This time on a Mac :) Lovin' it, @amygori 👍 

![screen shot 2016-06-16 at 7 48 49 pm](https://cloud.githubusercontent.com/assets/6612770/16136777/8936e3fe-33fb-11e6-9146-e6f9936986cd.png)

Roenok commented:
Yeah, all attempts to install and run failed catastrophically for me. I'm on a Windows 10 laptop. Errors included:
```Timed out while waiting for the machine to boot. This means that
Vagrant was unable to communicate with the guest machine within
the configured ("config.vm.boot_timeout" value) time period.

If you look above, you should be able to see the error(s) that
Vagrant had when attempting to connect to the machine. These errors
are usually good hints as to what may be wrong.

If you're using a custom box, make sure that networking is properly
working and you're able to connect to the machine. It is a common
problem that networking isn't setup properly in these boxes.
Verify that authentication configurations are also setup properly,
as well.

If the box appears to be booting properly, you may want to increase
the timeout ("config.vm.boot_timeout") value.```

```An error occurred in the underlying SSH library that Vagrant uses.
The error message is shown below. In many cases, errors from this
library are caused by ssh-agent issues. Try disabling your SSH
agent or removing some keys and try again.

If the problem persists, please report a bug to the net-ssh project.

timeout during server version negotiating```

```The following SSH command responded with a non-zero exit status.
Vagrant assumes that this means the command failed!

chown `id -u vagrant`:`id -g vagrant` /opt/gdi/development

Stdout from the command:

Stderr from the command:

stdin: is not a tty
chown: changing ownership of ‘/opt/gdi/development’: Not a directory```

Roenok commented:
Oh man, exciting developments in error messages up in here! Here is what I have discovered:
1. Your machine must support hardware virtualization. For some computers, including Lenovo Thinkpads, this means you must enable this at a BIOS level
2. You must have Git installed. In Windows, using Git within Powershell is not sufficient, as Vagrant cannot run properly within Powershell. Git Bash appears to work
3. Vagrant up will take a very long time to run.

Roenok commented:
1. Bonus note: MUST use the newest version of VirtualBox (at least above 5.0.12)

That seems to have been the secret sauce

amygori committed:
Vagrant provisioner cleanup

-- remove some duplication
-- annotate the intensely long time it takes to get Ruby and compile it
-- don't install ruby docs
-- don't try to bundle install in provisioner

amygori committed:
Add rad message of the day on vagrant ubuntu startup

kstack7 commented on motd.sh:
```diff
@@ -0,0 +1,13 @@
+(_)) __|(_) ((_)| |   |   \ (_))  _)((_)(_)) | | ((_)((_)_\  |_ _|| |_
+  | (_ || || '_|| |   | |) |/ -_) \ V / / -_)| |/ _ \| '_ \)  | | |  _|
+   \___||_||_|  |_|   |___/ \___|  \_/  \___||_|\___/| .__/  |___| \__|
+                                                     |_|
```
hehehe <3

amygori replied:
that cracks me up too
